install.sh

```diff
@@ -31,6 +31,18 @@ download() {
   fi
 }
 
+validate_url() {
+  url=$1
+
+  if command -v curl > /dev/null 2>&1; then
+    curl --output /dev/null --silent --head --fail "$url" >/dev/null 2>&1
+  else
+    wget --spider "$url" >/dev/null 2>&1
+  fi
+
+  return $?
```

Comment:
```suggestion
  if command -v curl > /dev/null 2>&1; then
    curl --output /dev/null --silent --show-error --location --head --fail "$url"
  else
    wget --spider --quiet "$url"
  fi
```

Reply:
* It would be better with curl to `--show-error` than to throw stderr away.
* It would be better with curl to follow redirects.
* `--output /dev/null` suffices. There is no need to redirect stdout to `/dev/null`.
* It would be better with wget to use the `--quiet` flag.
* The return code of the shell function will be the code of the final command. There is no need to `return $?`.

I think the error that curl and wget will print should be left intact so I suggest removing any file redirects.